lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,297 @@
+# frozen_string_literal: true
+
+require_relative '../../command'
+require 'tty-prompt'
+require 'fileutils'
+require 'erb'
+
+module Vtk
+  module Commands
+    class Socks
+      # Sets up socks access to the VA network
+      class Setup < Vtk::Command
+        attr_reader :ssh_config_path, :input, :output, :boot_script_path, :ssh_key_path, :prompt, :port, :skip_test
+
+        def initialize(options)
+          @options = options
+          @prompt = TTY::Prompt.new interrupt: :exit
+          @port = options['port'] || '2001'
+          @boot_script_path = options['boot_script_path'] || "#{ENV['HOME']}/Library"
+          @ssh_key_path = options['ssh_key_path'] || "#{ENV['HOME']}/.ssh/id_rsa_vagov"
+          @ssh_config_path = options['ssh_config_path'] || "#{ENV['HOME']}/.ssh/config"
+          @skip_test = options['skip_test'] || false
+
+          super()
+        end
+
+        def execute(input: $stdin, output: $stdout)
+          @input = input
+          @output = output
+
+          check_ssh_key
+          configure_ssh
+          test_ssh_connection unless skip_test
+          system_boot_configured = configure_system_boot
+          configure_system_proxy
+          test_http_connection
+          show_exit_message system_boot_configured
+        end
+
+        private
+
+        def launch_agent_label
+          @launch_agent_label ||= begin
+            launch_agent_label = 'gov.va.socks'
+            launch_agent_label += "-test-#{rand 1000}" if ENV['TEST'] == 'test'
+            launch_agent_label
+          end
+        end
+
+        def check_ssh_key
+          return true if key_exists?
+
+          open_key_access_request
+          exit
+        end
+
+        def key_exists?
+          File.exist? ssh_key_path
+        end
+
+        def open_key_access_request
+          url = 'https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?' \
+            'assignees=&labels=external-request%2C+operations&template=Environment-Access-Request-Template.md&' \
+            'title=Access+for+%5Bindividual%5D'
+          `#{macos? ? 'open' : 'xdg-open'} "#{url}"`
+          log "Please create an SSH key using `ssh-keygen`. You'll have to wait for access approval before " \
```

Comment:
I tried to run the `ssh-keygen` as a sub-process here but forwarding the stdin here proved extremely difficult. Try if you may but you've been warned.

Having the user run `ssh-keygen` manually and then re-run setup is easier.